packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -593,14 +593,15 @@ export interface OptionalFieldEditor extends FieldEditor<OptionalChangeset> {
 			fill: ChangesetLocalId;
 			detach: ChangesetLocalId;
 		},
+		revision: RevisionTag,
 	): OptionalChangeset;
 
 	/**
 	 * Creates a change which clears the field's contents (if any).
 	 * @param wasEmpty - whether the field is empty when creating this change
 	 * @param changeId - the ID associated with the detach.
 	 */
-	clear(wasEmpty: boolean, id: ChangesetLocalId): OptionalChangeset;
+	clear(wasEmpty: boolean, id: ChangesetLocalId, revision: RevisionTag): OptionalChangeset;
```

Comment:
Same idea as for `set`: take a `ChangeAtomId` instead of `ChangesetLocalId` + `RevisionTag`.